modules/diskq/diskq.c

```diff
@@ -180,6 +180,14 @@ _attach(LogDriverPlugin *s, LogDriver *d)
   if (self->options.prealloc < 0)
     self->options.prealloc = disk_queue_config_get_prealloc(cfg);
 
+  if (self->options.truncate_size_ratio < 1 && self->options.prealloc)
+    {
+      msg_error("Cannot disable disk-buffer truncation while preallocation is enabled. "
```

Comment:
I think the error message is incorrect, we should state the opposite here:
"Truncation can not be enabled together with preallocation. Please disable truncation if you want to use preallocation."

I think we should disable truncation for the user automatically in case they don't specify the truncate size ratio explicitly (that's probably the 90% of our use cases) but they enabled preallocation . We could show this error in case they explicitly enabled both features.

Reply:
Nice catch!

Fixed the warning and the commit message in the original commit.

Implementing the automatic disable was more complex, so I have done it in new commits. It is complex, because with the current implementation we cannot know if the values are set explicitly in the global options block. It is also complex, because we already have 3 different "default" `truncate-size-ratio()` values (set in 2 different places), absed on the config version. Anyways, I have implemented it, but we can drop it if you do not like it.

Reply:
Wow. Thank you :)